common/lwlibav_video.c

```diff
@@ -1250,8 +1250,10 @@ static inline int copy_field
         }
     }
     /* Treat this frame as interlaced. */
-    dst->interlaced_frame = src->interlaced_frame;
-    dst->top_field_first  = top_field_first;
+    if( !!(src->flags & AV_FRAME_FLAG_INTERLACED) )
```

Comment:
Need to check the false condition too.

```
else
    dst->flags &= ~AV_FRAME_FLAG_INTERLACED;
```

It's the same logic for the other two places too.